lib/services/local/headlessauthn_watcher.go

```diff
@@ -292,9 +300,13 @@ func (h *HeadlessAuthenticationWatcher) assignWaiter(ctx context.Context, name s
 		if h.waiters[i].ch != nil {
 			continue
 		}
-		h.waiters[i].ch = make(chan *types.HeadlessAuthentication)
-		h.waiters[i].name = name
+		h.waiters[i].ch = make(chan backend.Item)
+		h.waiters[i].key = string(headlessAuthenticationKey(name))
 		h.waiters[i].stale = make(chan struct{})
+
+		// initialize waiter as stale to force an initial backend check.
+		close(h.waiters[i].stale)
```

Comment:
How does the channel "heal" after marked as stale? Does it get assigned a fresh channel? Isn't that racy?

I wonder if making the channel buffered and sending something on it would work instead?

Reply:
Right now the stale channel is protected by the watcher's mutex, but I also like the idea of sending to a buffered channel instead.